x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/state.ts

```diff
@@ -32,6 +32,7 @@ export interface State {
     jobId: DataFrameAnalyticsId;
     jobIdExists: boolean;
     jobIdEmpty: boolean;
+    jobIdInvalidMaxLength?: boolean;
```

Comment:
Would be good to have this in line with the other attributes which are not optional. To fix it I think you just need to add a default `jobIdInvalidMaxLength: false;` to `getInitialState()`.